### DISCUSSION TITLE
Player events related question

### DISCUSSION THREAD

**Discussion by Tofaa2** [Minestom/Minestom#1253] [Help] [answered]
How would one go about finding an event that fires when a player changes slots of main hand or armor? I suppose i need to make them myself but I don't know where to start.

**Comment by VegetalDev** [comment, marked as answer]
`PlayerChangeHeldSlotEvent` to listen for selected item changes, `PlayerInventoryItemChangeEvent` to listen for any inventory change, including armor (you'll have to check the slot in your event handler).
